internal/serve/httphandler/verifiy_receiver_registration_handler.go

```diff
@@ -108,6 +108,12 @@ func (v VerifyReceiverRegistrationHandler) processReceiverVerificationPII(
 	now := time.Now()
 	truncatedPhoneNumber := utils.TruncateString(receiver.PhoneNumber, 3)
 
+	// Use the external_id associated with the receiver if it exists
+	if receiver.ExternalID != "" {
+		// Logic to handle the verification associated with the external_id
+		// This may include comparing the hashed phone number with the one associated with external_id
+	}
+	
```

Comment:
You'll probably want to do the externalID validation (and the hashed_phone_number validation) on or before the `Step 3` below:

https://github.com/stellar/stellar-disbursement-platform-backend/blob/977e0c359b76a61c6167fa30aff5b2e6912ed283/internal/serve/httphandler/verifiy_receiver_registration_handler.go#L136-L153